### PR TITLE
Issue / 9444 Subtitle Widget API Area Accept Component or Text as Parameter Value

### PR DESCRIPTION
**Pull request by 10upsimon**
## Summary

Addresses issue:

- #9444

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.
- [ ] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/googlesitekit/widgets/components/WidgetAreaHeader.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import { useSelect } from 'googlesitekit-data';
+import { CORE_WIDGETS } from '../datastore/constants';
+import WidgetNewBadge from './WidgetNewBadge';
+
+export default function WidgetAreaHeader( { slug } ) {
+	const widgetArea = useSelect( ( select ) =>
+		select( CORE_WIDGETS ).getWidgetArea( slug )
+	);
+
+	const { Icon, title, subtitle, CTA } = widgetArea;
```
Comment (zutigrm) [inline assets/js/googlesitekit/widgets/components/WidgetAreaHeader.js:37]: See the comments bellow, if we pass these as props directly, we do not need to re-import them from widget area selector, which is already handled in parent component

**Review by zutigrm — CHANGES_REQUESTED** [review]
Thanks @10upsimon looks good, I just left one main suggestion

File: assets/js/googlesitekit/widgets/components/WidgetAreaRenderer.js [hunk 2]
```diff
@@ -99,7 +99,7 @@ export default function WidgetAreaRenderer( { slug, contextID } ) {
 		select( CORE_WIDGETS ).getWidgetArea( slug )
 	);
 
-	const { Icon, title, style, subtitle, CTA, Footer } = widgetArea;
+	const { style, CTA, Footer } = widgetArea;
```
[the remark this answers is not in the document]
Comment (10upsimon) [reply assets/js/googlesitekit/widgets/components/WidgetAreaRenderer.js:102]: @zutigrm this sounds logical to me, I will update

[the remark this answers is not in the document]
Comment (10upsimon) [reply assets/js/googlesitekit/widgets/components/WidgetAreaRenderer.js]: @zutigrm addressed in the latest commit.

File: assets/js/googlesitekit/widgets/components/WidgetAreaHeader.js (hunk 1 above)
Comment (10upsimon) [reply assets/js/googlesitekit/widgets/components/WidgetAreaHeader.js]: @zutigrm addressed in the latest commit.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @10upsimon. Left some comments for you. Please, take a look.

**Comment by 10upsimon** [comment]
Thanks for the feedback @eugene-manuilov, I will address and return to you for re-review.

File: assets/js/googlesitekit/widgets/components/WidgetAreaRenderer.js [hunk 3]
```diff
@@ -197,40 +196,13 @@ export default function WidgetAreaRenderer( { slug, contextID } ) {
 					) }
 					ref={ widgetAreaRef }
 				>
-					<Row>
-						<Cell
-							className="googlesitekit-widget-area-header"
-							size={ 12 }
-						>
```
[the remark this answers is not in the document]
Comment (10upsimon) [reply assets/js/googlesitekit/widgets/components/WidgetAreaRenderer.js]: Thanks @eugene-manuilov this has been fixed in the latest set of commits.

File: assets/js/googlesitekit/widgets/components/WidgetAreaHeader.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+			</Cell>
+		</Row>
+	);
+}
+
+WidgetAreaHeader.propTypes = {
+	slug: PropTypes.string.isRequired,
+	Icon: PropTypes.bool,
+	title: PropTypes.string,
+	subtitle: PropTypes.oneOf( PropTypes.string, PropTypes.object ),
```
[the remark this answers is not in the document]
Comment (10upsimon) [reply assets/js/googlesitekit/widgets/components/WidgetAreaHeader.js]: Thanks @eugene-manuilov this has been fixed in the latest set of commits.

File: assets/js/googlesitekit/widgets/components/WidgetAreaHeader.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+				{ ( subtitle || CTA ) && (
+					<div className="googlesitekit-widget-area-header__details">
+						{ subtitle && (
+							<h4 className="googlesitekit-widget-area-header__subtitle">
+								{ typeof subtitle === 'function' ? (
+									<subtitle />
+								) : (
+									subtitle
+								) }
```
[the remark this answers is not in the document]
Comment (10upsimon) [reply assets/js/googlesitekit/widgets/components/WidgetAreaHeader.js]: Thanks @eugene-manuilov this has been fixed in the latest set of commits.

**Review by eugene-manuilov — APPROVED** [review]
Thanks, @10upsimon. LGTM.

**Pull request merged**